lib/modules/manager/gradle/extract.ts

```diff
@@ -52,7 +53,8 @@ export async function extractAllPackageFiles(
     };
 
     try {
-      const content = await readLocalFile(packageFile, 'utf8');
+      let content = await readLocalFile(packageFile, 'utf8');
+      content = content.replace(regEx(/\r\n/g), '\n');
```

Comment:
i still think this is wrong, the `updateDependency` function doesn't change those, so `fileReplacePosition` will be off by those removed chars

export function updateDependency({
  fileContent,
  upgrade,
}: UpdateDependencyConfig): string | null {
  if (upgrade.updateType === 'replacement') {
    logger.warn('maven manager does not support replacement updates yet');
    return null;
  }
  const offset = fileContent.indexOf('<');
  const spaces = fileContent.slice(0, offset);
  const restContent = fileContent.slice(offset);
  const updatedContent = updateAtPosition(restContent, upgrade, '</');
  if (!updatedContent) {
    return null;
  }
  if (updatedContent === restContent) {
    return fileContent;
  }
  return `${spaces}${updatedContent}`;
}

@zharinov WDYT?

Needs real runs to verify with windows newlines

Reply:
Yes, I think it'll break for some repos

Reply:
take a look at this:
![image](https://user-images.githubusercontent.com/42116482/166632240-c68f176b-a824-4659-b5aa-26c16697d7ff.png)

it doesn't reach the update functionality with `\r\n`, it gets removed before that.
well, if it fails before that, what should we do then?
i put it on the file read, it breaks everything
i put it once we get it to gradle resolver, you say it might break in the update, 
what should i do then?

Reply:
it might happen only if a repository has windows attributes in `.gitattributes` since we are reading file from the repository itself, so should i add it also to the global `update/branch/get-updated`?

or should i remove it from this task, create a refactor task and change it for all read files over all code?
xD i dont know what to do

Reply:
> it doesn't reach the update functionality with \r\n, it gets removed before that.

What is the reason it doesn't get updated with `\r\n`? Sounds like a bug to me.

Reply:
@zharinov  
in extraction phase, we Clone the repo, we take the file from our system, which is windows in my case

When we update, we take the file from the Repository itself, but by default, the repository is linux in all gitX i believe,
Unless you change the `.gitattributes` in the repository itself.

Reply:
Hmm, now I have two more questions:
1. In the `main` branch, does it lead to same error?
2. Is it possible to point to exact place in code where this bug finally causes `update` function not to be called? I.e. condition check or something...
🤔

Reply:
Anyway, I don't think Windows newline handling should be related to the original issue

Reply:
1) i am on the main branch , the file content is taken from : git/index.ts
```
 const content = await git.show([
      'origin/' + (branchName || config.currentBranch) + ':' + filePath,
    ]);
```

2) what do u mean, the update is always called it's another step completely.. 
the Extraction itself fails to find specific format of dependencies, and it doesnt extract at all.. which means there is no upgrade to do in the update functionality
see https://github.com/renovatebot/renovate/blob/main/docs/development/assets/renovate-flow.md

3) you are right, i will just remove this fix, even though it fixes it for windows. and i dont think it would break anything still, but we need a better task to handle windows issues because when we clone from windows files will be read with `\r\n` instead even if the repository is in github(linux)

Reply:
@PhilipAbed Until all windows line ending issues are fixed, you should set `core.autocrlf = input` on windows for your bot, so files will have linux endings by default if not overridden by `.gitattributes`

Reply:
do we have documentation/discussion for this?